superset/utils/core.py

```diff
@@ -580,6 +581,10 @@ def json_iso_dttm_ser(obj: Any, pessimistic: bool = False) -> str:
         return val
     if isinstance(obj, (datetime, date, pd.Timestamp)):
         obj = obj.isoformat()
+    elif isinstance(obj, psycopg2._range.DateTimeTZRange):
```

Comment:
Thanks for the fix, however database specific encoding should be handled by the associated DB engine spec. I _think_ the solution would be to move `json_iso_dttm_ser` to `db_engine_specs/base.py` and override it for PostgreSQL.